lib/javascript/StartupHelpers.js

```diff
@@ -61,20 +81,34 @@ module.exports.generateEntryStartup = (
 	 * @param {boolean=} final true when final, otherwise false
 	 */
 	const outputCombination = (chunks, moduleIds, final) => {
-		if (chunks.size === 0) {
+		const fn = runtimeTemplate.returningFunction(
+			moduleIds.map(runModule).join(", ")
+		);
+		const chunkIds = Array.from(chunks, c => c.id);
+		const params = {
+			chunkIds,
+			chunk,
+			fn,
+			final,
+			passive,
+			runtime,
+			EXPORT_PREFIX
+		};
+		const handled = hooks.generateFederationStartup.call(params);
```

Comment:
this hook never gets called it seems, the tap in AsyncStartupPlugin never triggers when tapping this hook? Maybe it shuold be part of javascript modules plugin? 
This will also need to be integrated into the esm plugin as well so that it covers all the places that generate startup code for entry